app/forms/select_placement_school_form.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class SelectPlacementSchoolForm
+  include ActiveModel::Model
+  include ActiveModel::AttributeAssignment
+  include ActiveModel::Validations::Callbacks
+  include Rails.application.routes.url_helpers
+  include ActionView::Helpers::TextHelper
+  include ActionView::Helpers::UrlHelper
+  include GovukLinkHelper
+
+  attr_accessor :slug, :school_id, :query, :trainee
+
+  validates :school_id, presence: true
+
+  def initialize(trainee:, query:, placement_slug: nil)
+    @trainee = trainee
+    @query = query
+    @slug = placement_slug
+  end
+
+  def search_results
+    @search_results ||= SchoolSearch.call(query:)
+  end
+
+  def form_path
+    if slug.present?
+      trainee_placement_school_search_path(trainee_id: trainee.slug, id: slug)
+    else
+      trainee_placement_school_search_index_path(trainee_id: trainee.slug)
+    end
+  end
+
+  def form_method
+    slug.present? ? :patch : :post
+  end
+
+  def title
```

Comment:
I'm not sure if this presentational stuff should be in here. Feels like it should go into a sep. view component/object? Maybe one for an enhancement!